portfolio/src/main/webapp/scripts/utilities.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Ensures that the stated arguments are present. 
+ * @param {...(NodeList|HTMLCollection|HTML element|string|Array)} args
+ *    Arguments to be checked.
+ */
+function ensureNonNull(... args) {
+  /**
+   * Indicates error type.
+   * @enum {string}
+   */
+  const ErrorType = {
+    NULL_VALUE: 'Null value',
+    EMPTY_STR: 'Empty string',
+    EMPTY_COLL_OR_LIST: 'Empty HTMLCollection/NodeList',
+    EMPTY_ARR: 'Empty array'
+  };
+
+  let hasError = false;
+  /** @type {ErrorType} The type of error, if any. */
+  let err = null;
+
+  for (arg of args) {
+    if (arg === null) {
+      err = ErrorType.NULL_VALUE;
+    } else if (typeof arg === 'string' && arg === '') {
+      err = ErrorType.EMPTY_STR;
+    } else if ((arg instanceof HTMLCollection || arg instanceof NodeList) &&
+        arg.length === 0) {
+          err= ErrorType.EMPTY_COLL_OR_LIST;
+    } else if (Array.isArray(arg) && arg.length === 0) {
+      err = ErrorType.EMPTY_ARR;
+    }
+
+    hasError = err !== null;
+    if (hasError) {
+      break;
+    }
+  }
+
+  console.assert(!hasError, `Missing desired element: ${err}`);
```

Comment:
Why do we need this `hasError` boolean flag?

I guess that simply doing the following would be enough?
```
if (error != null) {
 console.error(...);
 return;
}
```

Reply:
I guess, I thought that `hasError` sounds more intuitive.

Reply:
Why is it a `break` instead of `return`, shouldn't we early return if null is already found?

Reply:
Oh. I didn't think that it makes a difference. 😅 But yup, `return` sounds good (:

Reply:
Yep, as we don't long how long the list might be, an early return would always be good if it potentially saves some performance.